roles/matrix-base/templates/usr-local-bin/matrix-remove-all.j2

```diff
@@ -15,11 +15,26 @@ if [ "$sure" != "Yes, I really want to remove everything!" ]; then
 	exit 0
 else
 	echo "Stop and remove matrix services"
-	for s in $(find {{ matrix_systemd_path }}/ -name "matrix-*" -printf "%f\n"); do
+
+	# Look for and stop services, avoiding things like
+	# 'matrix-synapse-worker@.service' (just a template for instantiated services; can't stop it directly).
+	# We use '-xtype f' and not '-type f', because we wish to match symlinks like this:
+	# '/etc/systemd/system/matrix-synapse.service.wants/matrix-synapse-worker@generic_worker:18111.service'
+	# and stop these instantiated services as well.
+	for s in $(find {{ matrix_systemd_path }}/ -xtype f -name "matrix-*" -printf "%f\n" | grep -v '@.service' | uniq); do
 		systemctl stop $s
+	done
+
+	# Get rid of regular service files, as well as symlinks like
+	# '/etc/systemd/system/matrix-synapse.service.wants/matrix-synapse-worker@generic_worker:18111.service'
+	# and even
+	# '/etc/systemd/system/multi-user.target.wants/matrix-synapse.service'.
+	for s in $(find {{ matrix_systemd_path }}/ -xtype f -name "matrix-*" -printf "%p\n"); do
```

Comment:
Maybe do `systemctl disable $service` first before cleaning up stuff with `rm`, ideally that should remove most symlinks already and is cleaner. If disabling services is not cleaning up everything, i'd get out the `rm`...

Reply:
I've switched to `systemctl disable --now $s` in c05d3d09bd22, but it still seems like we need to `rm` symlinks, as explained in that commit.

Reply:
`disable`ing atleast tells systemd to expect that this unit is gone, and should remove it from `multi-user.target.wants`. If specified in-spec, it should (afaik) also be possible to be auto-removed from `matrix-synapse.service.wants`.

Reply:
True.

We are specifying them in `Wants=..` for `matrix-synapse.service`. Not sure why systemd doesn't auto-remove them from the directory.

Reply:
This has been redone. Now that we don't use systemd instantiated services and don't have `[Install] WantedBy=...` clauses in workers, things are simpler and we're back to something similar to what we had before:

```sh
	for s in $(find {{ matrix_systemd_path }}/ -type f -name "matrix-*" -printf "%f\n"); do
		systemctl disable --now $s
		rm -f {{ matrix_systemd_path }}/$s
	done
```